compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -1796,6 +1796,33 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{output_shape};
   }
 
+  loco::NodeShape visit(const luci::CircleWhere *node) final
+  {
+    assert(node->arity() > 0);
+
+    // TODO for arity 1
```

Comment:
Is arity 1 codes going to be included in this draft?

Reply:
I have a problem with shape deduction if arity = 1. Because it depends оn num_true (runtime value). Perhaps you will tell me standard way for this case.

Reply:
Not standard and still ambigious but in this case I've used like this one.
```cxx
    loco::TensorShape shape = own_shape(node);
    return loco::NodeShape{shape};
```
It would be nice if you add a comment like
```cxx
// NOTE when numValues() == 1, we can not deduce shape as it depends on num_true(runtime_value)
```

Reply:
I suppose `loco::TensorShape shape = own_shape(node);` will work only if shape is set in importer (or something like this), but one-input form of this operator is actually can be resolved only during execution.

This also could be a pain for interpreter, I need to talk about this operator with @s-barannikov.

Reply:
1) There are similar cases when import has unknown shape. @s-barannikov may know about this.
2) There are already some Ops do this
3) test codes don't set shape when creation

I've revised `CircleShapeInferenceRule::infer()` to handle this but not fully comforatable...